testing/adios2/engine/staging-common/TestCommonReadAttrs.cpp

```diff
@@ -293,16 +293,19 @@ TEST_F(CommonReadTest, ADIOS2CommonRead1D8)
         TimeGapDetected++;
     }
 
-    if (TimeGapExpected)
-    {
-        EXPECT_TRUE(TimeGapDetected);
-    }
-    else
+    if (!IgnoreTimeGap)
     {
+        if (TimeGapExpected)
+        {
+            EXPECT_TRUE(TimeGapDetected);
+        }
+        else
+        {
+            EXPECT_FALSE(TimeGapDetected);
+        }
         EXPECT_EQ(t, NSteps);
```

Comment:
Would you move the "EXPECT_EQ(t, NSteps)" outside the "if (!IgnoreTimeGap)"  to make sure we're checking that we got all the expected steps (even if we're ignoring writer side test delays).